### DISCUSSION TITLE
Airflow helm configs not being exposed on UI when we follow the documentation

### DISCUSSION THREAD

**Discussion by kunaljubce** [apache/airflow#38679] [General]
For airflow:2.8.2, we are doing a local k8s installation and setup of Airflow using official helm chart. When I start the webserver with the defaults, and navigate to the Admin --> Configurations tab, it shows as "Your Airflow administrator chose not to expose the configuration, most likely for security reasons."

![image](https://github.com/apache/airflow/assets/34453918/398a2009-d634-481d-9d53-dd8cf30054d3)

Following the [documentation](https://airflow.apache.org/docs/helm-chart/stable/airflow-configuration.html#configuring-airflow), when I add this config, I still see the same error. Validated the same config inside the webserver pod itself and it seems this does not infact set the config to True. 

```
config:
  webserver:
    expose_config: 'True'  # by default this is 'False'
```

<img width="1480" alt="image" src="https://github.com/apache/airflow/assets/34453918/641bd2a6-9db3-4161-a5da-4f1e2468b75b">

However, when I do the following - 
```
env:
  - name: "AIRFLOW__WEBSERVER__EXPOSE_CONFIG"
  value: "True
```
I see that the config actually works and the configs are actually visible on UI - 

<img width="1480" alt="image" src="https://github.com/apache/airflow/assets/34453918/cee4c4e5-41c7-48d1-a88b-07431e16255c">

![image](https://github.com/apache/airflow/assets/34453918/e9200029-e71e-4c51-9559-8a385cbe68ba)

Do we need to get this documentation issue fixed?

**Comment by potiuk** [comment]
If you find out why then feel free.

**Reply by kunaljubce** [reply]
@potiuk Why as in, find out why it is behaving as I mentioned? Honestly I have not looked into the airflow codebase at all, could you let me know where to start?

**Reply by potiuk** [reply]
The answer is https://github.com/apache/airflow/tree/main/chart

And maybe someone will be able to answer your question without you doing, it as well. I am just not sure I am able to jump on all questions here and do all the digging in the code - because I do not know the answer by heart - so I propose you do it. But again - maybe someone else will have time to dig in and answer. The proposal to dig-in and try to understand what's going on was because it's open-source and anyone can do it.

**Reply by kunaljubce** [reply]
Thanks, closing this discussion!

I started looking into this and it seems like I might have missed something in my initial trial itself. The way it is mentioned in the documentation works just fine - 

```
config:
  webserver:
    expose_config: 'True'  # by default this is 'False'
```
